agent-inject/agent/agent.go

```diff
@@ -383,8 +383,22 @@ func (a *Agent) Patch() ([]byte, error) {
 			"/spec/volumes")...)
 	}
 
-	//Add Volume Mounts
+	//Add Volume Mounts to desired containers
+	raw, ok := a.Pod.Annotations[AnnotationAgentInjectContainers];
+	if !ok {
+		return patches, fmt.Errorf("vault.hashicorp.com/agent-inject-containers annotation not found")
+	}
+
+	names := make(map[string]struct{})
+	for _, name := range strings.Split(raw, ",") {
+		names[name] = struct{}{}
+	}
```

Comment:
I'm thinking this code isn't necessary because if we could extend the `Agent` struct to have a `Containers []string` type, then populate that using the [Init function](https://github.com/hashicorp/vault-k8s/pull/163/files#diff-1a88ba1e5c2e3b5343b2a2f07199c01eR334).  If the slice is empty, apply to all containers, if it's not, only mount to containers that match.